presto-main/src/main/java/com/facebook/presto/operator/scalar/StringFunctions.java

```diff
@@ -617,6 +617,35 @@ public static Slice rightPad(@SqlType("varchar(x)") Slice text, @SqlType(Standar
         return pad(text, targetLength, padString, text.length());
     }
 
+    @Description("computes Levenshtein Distance between two strings")
+    @ScalarFunction
+    @LiteralParameters({"x", "y"})
+    @SqlType(StandardTypes.BIGINT)
+    public static long levenshteinDist(@SqlType("varchar(x)") Slice string1, @SqlType("varchar(y)") Slice string2)
+    {
+        int[] codePoints1 = castToCodePoints(string1);
+        int[] codePoints2 = castToCodePoints(string2);
+        long[] prevDistances = new long[codePoints2.length + 1];
+        long[] currDistances = new long[codePoints2.length + 1];
+
+        for (int j = 0; j < codePoints2.length + 1; ++j) {
```

Comment:
It would be better to use `i` as iterating variable, and use `i++` instead of `++i`, which will make the code easier to read for most people.